samples/jwt-validator/README.md

```diff
@@ -0,0 +1,109 @@
+# JSON Web Token (JWT) Validator
+
+This folder contains a Spin application which is able to validate JWT tokens issued by any OAuth 2.0 & OpenID Connect compliant Identity Provider (or Token Issuer). 
+
+For demonstration purposes, the Spin app responds with a HTTP status code `200` if the token is valid. Otherwise, it returns a `401` and provides detailed information about what is invalid as the response payload. In a real-world setting, this could be combined with request teeing and forward requests to the actual origin, if the presented token is valid.
+
+## Exposed Endpoints
+
+The application exposes two endpoints:
+
+- `POST /validate`: Which is using a pre-configured validation setting
+- `POST /validate-with-options`: Which accepts validation options as JSON payload 
+
+Invoking the `/validate` endpoint (and presenting a JWT using the standard `Authorization` header), the following aspects of the JWT token are validated:
+
+- Correct JWT format
+- Ensure JWT integrity
+- Ensure token is not expired
+- Ensure token is already valid (NBF)
+
+## Optional Validation
+
+- Validating Token Types
+- Validating Token Issuer
+- Validating audiences
+- Validating scopes
+- Validating claim existence
+
+
+## Demo Flow
+
+### Requesting JWT Tokens
+
+```console
+# requesting a token using the customer-client
+customer_token=$(curl -H 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'grant_type=client_credentials' \
+--data-urlencode 'client_id=customer-client' \
+--data-urlencode 'client_secret=511536EF-F270-4058-80CA-1C89C192F69A' \
+--data-urlencode 'scope=customer.read' https://idsrv.purplesky-721836c2.eastus.azurecontainerapps.io/connect/token | jq -r .access_token)
+
+# requesting a token using the invoice-client
+invoice_token=$(curl -H 'Content-Type: application/x-www-form-urlencoded' \
+--data-urlencode 'grant_type=client_credentials' \
+--data-urlencode 'client_id=invoice-client' \
+--data-urlencode 'client_secret=511536EF-F270-4058-80CA-1C89C192F69A' \
+--data-urlencode 'scope=invoice.read' https://idsrv.purplesky-721836c2.eastus.azurecontainerapps.io/connect/token | jq -r .access_token)
+```
+
+### Compiling the Spin application
+
+```console
+spin build
```

Comment:
I'm seeing the following when I `spin build`:
```
   Compiling ring v0.17.8
warning: ring@0.17.8: error: unable to create target: 'No available targets are compatible with triple "wasm32-unknown-wasip1"'
warning: ring@0.17.8: 1 error generated.
error: failed to run custom build command for `ring v0.17.8`
```

With: `spin 3.4.0-pre0 (0e404a42 2025-07-15)` and `rustc 1.88.0 (6b00bc388 2025-06-23)`.

I've attempted to install that target, but I get `error: toolchain 'stable-aarch64-apple-darwin' does not support target 'wasm32-unknown-wasip1'`.  Maybe I need to use a different toolchain?

Reply:
The rust target changed to `wasm32-wasip2`

Reply:
`--target wasm32-wasip1` <-- sneaky 1 vs 2

Reply:
They have a test for it: https://github.com/briansmith/ring/blob/b9c8f4f530d84df1aec63a83f51dbaa8078c7e59/.github/workflows/ci.yml#L187

I wonder where `unknown` is coming from :thinking: